src/Domain/CoordinateMaps/TimeDependent/Shape.hpp

```diff
@@ -83,16 +84,14 @@ namespace domain::CoordinateMaps::TimeDependent {
  * `jacobian` currently calculates the `interpolation_info` with an order
  * higher.
  */
-template <typename TransitionFunc>
```

Comment:
I think a case could be made for keeping the shape map templated on the transition func, since I'm pretty sure the choice of transition func can always be made at compile time. Some annoyances with pointers would disappear, but other annoyances with templates would arise. Not sure how relevant this is for performance. I'll leave the decision up to you, but please add a "design decisions" comment about this to the docs.